pkg/gang/gangclient/gangclient.go

```diff
@@ -0,0 +1,84 @@
+/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func main() {
+	jobName := os.Getenv("jobName")
+	serviceName := os.Getenv("serviceName")
+	memberAmount, err := strconv.Atoi(os.Getenv("memberAmount"))
+	if err != nil {
+		memberAmount = 10
+	}
+	runtimeMin, err := strconv.Atoi(os.Getenv("runtimeMin"))
+	if err != nil {
+		runtimeMin = 1
+	}
+	serviceName = strings.ToUpper(serviceName) + "_SERVICE_HOST"
+	serviceIP := os.Getenv(serviceName)
+	addRequest(serviceIP, jobName)
+	// Check if we satisfy gang minMember or not every 2 second
+	for {
+		number := checkRequest(serviceIP, jobName)
+		if number >= memberAmount {
+			fmt.Println("satisfy gang minMember.")
+			fmt.Println("start to run job.")
+			time.Sleep(time.Duration(runtimeMin) * time.Minute) // means the application start running job
+			break
+		}
+		time.Sleep(2 * time.Second)
```

Comment:
if the condition is not met, we need to print some logs as well.
E.g how many members are running out there, and what is the minMember, can you add that log as well?

Reply:
done